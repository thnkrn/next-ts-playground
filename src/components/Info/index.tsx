import { styled } from '@compiled/react'
import TYPO from 'constants/typography'

const StyledContainer = styled.div`
  @media screen and (min-width: 481px) and (max-width: 1024px) {
    margin: 40px 20px;
  }

  @media screen and (max-width: 480px) {
    margin: 40px 0 !important;
  }

  margin: 60px 120px;
`

const StyledWrapper = styled.div`
  display: flex;
  flex-direction: column;
  white-space: pre-line;
  text-align: justify;

  .body2 {
    :not(:first-of-type) {
      margin-top: 10px;
    }
  }
`

const Info = () => (
  <StyledContainer>
    <StyledWrapper>
      <span className={TYPO.body2} style={{ color: 'var(--thnkrn-blue)' }}>
        Cheers,
      </span>
      <span className={TYPO.body2}>
        I'm a Software Engineer who passionate about web development or you can call me 'Webmaster'
        which is based on React ecosystem. I'm working on a fancy landing page, SEO, security,
        performance, DevOps, and followed by the optimization part that I'm mostly crazy about it.
        And now as a Fullstack Developer, I'm also working on the Backend stuff, and Database too.
      </span>
      <span className={TYPO.body2}>
        Moreover, I'm usually opened and interested in new tech stacks and I would like to try it
        ASAP. I'm also free to listen and discuss the new tech stacks and how it is awesome with my
        teammates.
      </span>
      <span className={TYPO.body2}>
        Lastly, I'm a fast learner and open to every avenue that comes to me. And I'm eager to learn
        new things very much!
      </span>
    </StyledWrapper>
  </StyledContainer>
)

export default Info
