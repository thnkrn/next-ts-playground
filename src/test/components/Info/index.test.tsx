import Info from 'components/Info'
import { render } from 'test/util'

describe('Info', () => {
  describe('Rendering', () => {
    it('should match snapshot', () => {
      const { asFragment } = render(<Info />)

      expect(asFragment()).toMatchSnapshot()
    })
  })
})
