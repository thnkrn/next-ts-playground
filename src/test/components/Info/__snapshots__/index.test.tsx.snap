// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info Rendering should match snapshot 1`] = `
<DocumentFragment>
  <style
    data-cmpld="true"
  >
    ._19pk1wqb{margin-top:60px}._2hwxllbu{margin-right:120px}._otyr1wqb{margin-bottom:60px}._18u0llbu{margin-left:120px}@media screen and (min-width:481px) and (max-width:1024px){._d7rp1ylp{margin-top:40px}._12e9gktf{margin-right:20px}._wnlt1ylp{margin-bottom:40px}._17npgktf{margin-left:20px}}@media screen and (max-width:480px){._za9y1fu0{margin-top:40px!important}._99s2grf3{margin-right:0!important}._183z1fu0{margin-bottom:40px!important}._wf1ngrf3{margin-left:0!important}}
  </style>
  <div
    class="_d7rp1ylp _12e9gktf _wnlt1ylp _17npgktf _za9y1fu0 _99s2grf3 _183z1fu0 _wf1ngrf3 _19pk1wqb _2hwxllbu _otyr1wqb _18u0llbu"
  >
    <style
      data-cmpld="true"
    >
      ._1e0c1txw{display:flex}._2lx21bp4{flex-direction:column}._o5726vc6{white-space:pre-line}._y3gncvlb{text-align:justify}._pn1d19bv .body2:not(.body2:first-of-type){margin-top:10px}
    </style>
    <div
      class="_1e0c1txw _2lx21bp4 _o5726vc6 _y3gncvlb _pn1d19bv"
    >
      <span
        class="body2"
      >
        Cheers,
      </span>
      <span
        class="body2"
      >
        I'm a Software Engineer who works as a full-stack developer. I'm passionate about web development based on React ecosystem and have a deep love for the NextJS framework. Furthermore, I'm coding on the Golang to create the backend enterprise-grade system by applying clean architecture.
      </span>
      <span
        class="body2"
      >
        I'm paying attention the most to developer experience like writing clean, composable, and maintainable code as world-class products for both internal and external customers. Also, I really care about the processes, high quality of code and products, and aesthetics.
      </span>
      <span
        class="body2"
      >
        Moreover, I'm a tech-savvy guy, who is always curious and excite about new technology stacks. I'm also like to learn, listen, and discuss tech things with my teammates.
      </span>
      <span
        class="body2"
      >
        Lastly, I'm a fast learner and open to every avenue that comes to me. And I'm eager to learn new things very much!
      </span>
    </div>
  </div>
</DocumentFragment>
`;
